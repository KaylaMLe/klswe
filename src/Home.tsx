import { ReactElement, useState } from 'react';
import { css } from '@emotion/css';
import chime from './assets/audio/chime.mp3';
import flower from './assets/images/flower.png';
import tree from './assets/images/tree.png';
import voloff from './assets/images/voloff.png';
import volon from './assets/images/volon.png';
import { keyframes } from '@emotion/react';


interface FlowerProps {
	id: number;
	dX: number;
	dY: number;
	rotation: number;
};

export function Home(): ReactElement {
	const [muted, setMuted] = useState(false);
	const homeStyle: string = css({
		boxSizing: 'inherit',
		height: '90vh',
		minHeight: '375px',
		display: 'flex',
		alignItems: 'center',
		justifyContent: 'center',
		position: 'relative',
	});

	return (
		<div className={homeStyle}>
			<MuteBtn onClick={() => { setMuted(!muted) }} muted={muted} />
			<Tree muted={muted} />
		</div>
	);
}

function MuteBtn({ onClick, muted }: { onClick: () => void, muted: boolean }): ReactElement {
	const muteBtnStyle = css({
		cursor: 'pointer',
		position: 'absolute',
		top: '5px',
		left: '5px',
		width: '200px',
		fontSize: '11pt',
		backgroundColor: '#EEA8D9',
		borderRadius: '2rem',
		borderStyle: 'solid',
		borderColor: '#602267',
		padding: '8px',
		display: 'flex',
		alignItems: 'center',
		justifyContent: 'space-between',
	});

	return (
		<button className={muteBtnStyle} onClick={onClick}>
			<img src={muted ? volon : voloff} alt='Mute all SFX' />
			{muted ? 'Unmute sound effects' : 'Mute sound effects'}
		</button>
	);
}

function Tree({ muted }: { muted: boolean }): ReactElement {
	const [flowerProps, setFlowerProps] = useState<FlowerProps[]>([]);
	const [flowerId, setFlowerId] = useState(0);

	const treeBoxStyle: string = css({
		cursor: 'pointer',
		position: 'relative',
	});

	const addFlower = () => {
		if (flowerProps.length === 64) {
			setFlowerProps(flowerProps.slice(1));
		}

		setFlowerProps([...flowerProps,
		{
			id: flowerId,
			// flower can be positioned across entire width of tree (258 px)
			dX: Math.floor(Math.random() * 259),
			// flower height is limited to branches (top 169 px)
			dY: Math.floor(Math.random() * 170),
			rotation: Math.floor(Math.random() * 360),
		}]);

		setFlowerId((flowerId + 1) % 64);

		if (!muted) {
			const audioElement = document.getElementById('clickSound') as HTMLAudioElement;

			if (audioElement) {
				audioElement.play();
			}
		}
	};

	return (
		<div className={treeBoxStyle} onClick={addFlower}>
			<audio id="clickSound">
				<source src={chime}></source>
				Your browser does not support the audio element.
			</audio>
			<img src={tree} alt='A drawing of a bonsai tree' />
			{flowerProps.map(props => (
				<Flower key={props.id} id={props.id} dX={props.dX} dY={props.dY} rotation={props.rotation} />
			))}
		</div>
	);
}

function Flower({ id, dX, dY, rotation }
	: { id: number, dX: number, dY: number, rotation: number }): ReactElement {
	const spinAnimation = keyframes`
		from {
			rotate: 0deg;
		}
		to {
			rotate: ${rotation}deg;
		}
	`;

	const flowerStyle: string = css({
		position: 'absolute',
		left: `${dX}px`,
		top: `${dY}px`,
		rotate: `${rotation}deg`,
		'@media(prefers-reduced-motion: no-preference)': {
			animation: `${spinAnimation} 0.5s ease-in-out`
		},
	});

	return (
		<img
			id={id.toString()}
			className={flowerStyle}
			src={flower}
			alt='A drawing of a pink flower'
		/>
	);
}