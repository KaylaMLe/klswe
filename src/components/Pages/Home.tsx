import { css, keyframes } from '@emotion/css';
import React, { useState } from 'react';
import { NavBar } from '../NavBar/NavBar';
import chime from '../../assets/audio/chime.wav';
import flower from '../../assets/images/flower.png';
import tree from '../../assets/images/tree.png';
import voloff from '../../assets/images/voloff.png';
import volon from '../../assets/images/volon.png';


interface FlowerProps {
	id: number;
	dX: number;
	dY: number;
	rotation: number;
};

export function Home(): React.JSX.Element {
	const [muted, setMuted] = useState(false);

	const pageStyle = css({
		height: '100vh',
	});
	const homeStyle = css({
		boxSizing: 'border-box',
		display: 'flex',
		alignItems: 'center',
		justifyContent: 'center',
		height: '90vh',
		minHeight: '375px',
		position: 'relative',
	});

	return (
		<div className={pageStyle}>
			<NavBar />
			<div className={homeStyle}>
				<MuteBtn onClick={() => { setMuted(!muted) }} muted={muted} />
				<Tree muted={muted} />
			</div>
		</div>
	);
}

function MuteBtn({ onClick, muted }: { onClick: () => void, muted: boolean }): React.JSX.Element {
	const muteBtnStyle = css({
		backgroundColor: '#EEA8D9',
		borderColor: '#602267',
		borderRadius: '2rem',
		borderStyle: 'solid',
		display: 'flex',
		alignItems: 'center',
		justifyContent: 'space-between',
		fontSize: '11pt',
		padding: '8px',
		position: 'absolute',
		top: '10px',
		left: '5px',
		width: '200px',
		zIndex: '1',
	});

	return (
		<button id='main' className={muteBtnStyle} onClick={onClick}>
			<img src={muted ? volon : voloff} alt='Toggle sound effects' />
			{muted ? 'Unmute sound effects' : 'Mute sound effects'}
		</button>
	);
}

function Tree({ muted }: { muted: boolean }): React.JSX.Element {
	const [flowerProps, setFlowerProps] = useState<FlowerProps[]>([]);
	const [flowerId, setFlowerId] = useState(0);

	const treeBoxStyle = css({
		cursor: 'pointer',
		position: 'relative',
	});

	const addFlower = () => {
		if (flowerProps.length === 64) {
			setFlowerProps(flowerProps.slice(1));
		}

		setFlowerProps([...flowerProps,
		{
			id: flowerId,
			// flower can be positioned across entire width of tree (258 px)
			dX: Math.floor(Math.random() * 259),
			// flower height is limited to branches (top 169 px)
			dY: Math.floor(Math.random() * 170),
			rotation: Math.floor(Math.random() * 360),
		}]);

		setFlowerId((flowerId + 1) % 64);

		if (!muted) {
			const audioElement = document.getElementById('clickSound') as HTMLAudioElement;

			if (audioElement) {
				audioElement.play();
			}
		}
	};

	const handleKeyDown = (event: React.KeyboardEvent) => {
		if (event.key === 'Enter' || event.key === ' ') {
			addFlower();
		}
	};

	return (
		<div
			className={treeBoxStyle}
			tabIndex={0}
			onClick={addFlower}
			onKeyDown={handleKeyDown}
		>
			<audio id="clickSound">
				<source src={chime}></source>
				Your browser does not support the audio element.
			</audio>
			<img src={tree} alt='A drawing of a bonsai tree' />
			{flowerProps.map(props => (
				<Flower key={props.id} id={props.id} dX={props.dX} dY={props.dY} rotation={props.rotation} />
			))}
		</div>
	);
}

function Flower({ id, dX, dY, rotation }
	: { id: number, dX: number, dY: number, rotation: number }): React.JSX.Element {
	const spinAnimation = keyframes`
		from {
			rotate: 0deg;
		}
		to {
			rotate: ${rotation}deg;
		}
	`;

	const flowerStyle = css({
		position: 'absolute',
		left: `${dX}px`,
		top: `${dY}px`,
		rotate: `${rotation}deg`,
		'@media(prefers-reduced-motion: no-preference)': {
			animation: `${spinAnimation} 0.5s ease-in-out`
		},
	});

	return (
		<img
			id={id.toString()}
			className={flowerStyle}
			src={flower}
			alt='A drawing of a pink flower'
		/>
	);
}