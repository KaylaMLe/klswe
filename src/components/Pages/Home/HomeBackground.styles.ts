import { CSSObject, keyframes } from '@emotion/react';

export const gradientWrapperStyle: CSSObject = {
	position: 'absolute',
	top: 0,
	left: 0,
	width: '100%',
	height: '100%',
	overflow: 'hidden',
	zIndex: -2,
};

export const gradientBackgroundStyle: CSSObject = {
	position: 'absolute',
	top: 0,
	left: 0,
	width: '400%',
	height: '100%',
	backgroundSize: '400% 100%',
};

export const sweepGradientKeyframes = keyframes`
0% {
	background-position: 400% 50%;
}
100% {
	background-position: 0% 50%;
}
`;

export const blueSweepStyle: CSSObject = {
	background: 'linear-gradient(120deg, transparent 0%, transparent 20%,rgb(12, 0, 67) 50%, transparent 80%, transparent 100%)',
	animation: `${sweepGradientKeyframes} 90s linear infinite`,
};

export const purpleSweepStyle: CSSObject = {
	background: 'linear-gradient(210deg, transparent 0%, transparent 25%,rgba(34, 0, 67, 0.65) 50%, transparent 75%, transparent 100%)',
	animation: `${sweepGradientKeyframes} 100s linear infinite reverse`,
};

export const starStyle: CSSObject = {
	position: 'absolute',
	backgroundColor: 'white',
	borderRadius: '50%',
	opacity: 0,
	transition: 'opacity 1s ease-in-out',
	boxShadow: '0 0 4px rgba(255, 255, 255)',
};

export const starBoxStyle: CSSObject = {
	position: 'absolute',
	top: '0px',
	left: '0px',
	width: '100vw',
	height: '105vmin',
	maxHeight: '100vh',
	overflow: 'hidden',
	zIndex: -1,
};
