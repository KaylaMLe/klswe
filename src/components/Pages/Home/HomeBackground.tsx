/** @jsxImportSource @emotion/react */
import React, { useState, useEffect, useRef } from 'react';
import {
	blueSweepStyle,
	gradientBackgroundStyle,
	gradientWrapperStyle,
	purpleSweepStyle,
	starBoxStyle,
	starStyle,
} from './HomeBackground.styles';

export function GradientSweeps(): React.JSX.Element {
	return (
		<div css={gradientWrapperStyle}>
			<div css={{ ...blueSweepStyle, ...gradientBackgroundStyle }} />
			<div css={{ ...purpleSweepStyle, ...gradientBackgroundStyle }} />
		</div>
	);
}

interface Star {
	id: number;
	x: number;
	y: number;
	opacity: number;
	size: number;
}

export function StarBox(): React.JSX.Element {
	const [stars, setStars] = useState<Record<number, Star>>({});
	const [starCount, setStarCount] = useState(0);
	const [maxStars, setMaxStars] = useState(100);
	const [isTabVisible, setIsTabVisible] = useState(!document.hidden);
	const containerRef = useRef<HTMLDivElement>(null);
	const nextId = useRef(0);

	// Calculate max stars based on starBox dimensions
	const calculateMaxStars = () => {
		const starBoxWidth = window.innerWidth;
		const starBoxHeight = Math.min(
			window.innerWidth * 1.05, // 105vmin (innerHeight * 1.05 is always greater than 100vh)
			window.innerHeight // cap at 100vh
		);

		const area = starBoxWidth * starBoxHeight;
		const starsPerArea = area / 22500;
		return Math.floor(starsPerArea);
	};

	// Generate weighted random star size (70% 1px, 30% 2px)
	const generateStarSize = () => {
		const random = Math.random();
		return random < 0.7 ? 1 : 2;
	};

	const createStar = () => {
		if (!containerRef.current || !isTabVisible) return;

		const container = containerRef.current;
		const x = Math.random() * container.offsetWidth;
		const y = Math.random() * container.offsetHeight;

		const newStar: Star = {
			id: nextId.current++,
			x,
			y,
			opacity: 0,
			size: generateStarSize(), // Weighted random size
		};

		setStars((prev) => ({ ...prev, [newStar.id]: newStar }));
		setStarCount((prev) => prev + 1);

		// star created on our end -> star rendered/added to dom
		requestAnimationFrame(() => {
			// star rendered -> star fade in
			requestAnimationFrame(() => {
				setStars((prev) => ({
					...prev,
					[newStar.id]: { ...prev[newStar.id], opacity: 1 },
				}));
			});
		});
	};

	const removeStar = () => {
		if (!isTabVisible) return;

		setStars((prev) => {
			const starIds = Object.keys(prev).map(Number);
			if (starIds.length === 0) return prev;

			const randomId = starIds[Math.floor(Math.random() * starIds.length)];

			// Fade out the star
			const updatedStars = { ...prev };
			updatedStars[randomId] = { ...updatedStars[randomId], opacity: 0 };

			// Remove the star after fade out completes
			setTimeout(() => {
				// Check if tab is still visible before removing
				if (isTabVisible) {
					setStars((current) => {
						const { [randomId]: removed, ...remaining } = current;
						return remaining;
					});
					setStarCount((current) => current - 1);
				}
			}, 1000);

			return updatedStars;
		});
	};

	// Initialize max stars and handle window resize
	useEffect(() => {
		const updateMaxStars = () => {
			const newMaxStars = calculateMaxStars();
			setMaxStars(newMaxStars);
		};

		// Set initial max stars
		updateMaxStars();

		// Add resize listener
		window.addEventListener('resize', updateMaxStars);

		return () => {
			window.removeEventListener('resize', updateMaxStars);
		};
	}, []);

	// Handle page visibility changes
	useEffect(() => {
		const handleVisibilityChange = () => {
			const isVisible = !document.hidden;
			setIsTabVisible(isVisible);
		};

		document.addEventListener('visibilitychange', handleVisibilityChange);

		return () => {
			document.removeEventListener('visibilitychange', handleVisibilityChange);
		};
	}, []);

	useEffect(() => {
		// Only run star operations when tab is visible
		if (!isTabVisible) return;

		const createInterval = setInterval(() => {
			if (starCount < maxStars) {
				createStar();
			}
		}, Math.random() * 1000 + 500); // Random between 500-1500ms

		const removeInterval = setInterval(() => {
			const minStars = Math.floor(maxStars * 0.8); // 80% of max stars

			if (starCount > minStars) {
				removeStar();
			}
		}, Math.random() * 1000 + 500);

		return () => {
			clearInterval(createInterval);
			clearInterval(removeInterval);
		};
	}, [starCount, maxStars, isTabVisible]);

	return (
		<div ref={containerRef} css={starBoxStyle}>
			{Object.values(stars).map((star) => (
				<div
					key={star.id}
					css={starStyle}
					style={{
						left: `${star.x}px`,
						top: `${star.y}px`,
						opacity: star.opacity,
						width: `${star.size}px`,
						height: `${star.size}px`,
					}}
				/>
			))}
		</div>
	);
}
