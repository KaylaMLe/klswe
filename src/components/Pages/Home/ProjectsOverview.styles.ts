import { CSSObject } from '@emotion/react';

export const overviewContainerStyle: CSSObject = {
	display: 'flex',
	flexDirection: 'column',
	alignItems: 'flex-start',
	width: '90vw',
	maxWidth: '100vw',
	margin: '2rem auto',
	fontFamily: 'Quicksand, "Arial Rounded MT Light", var(--font-sans-serif)',
};

export const overviewTitleStyle: CSSObject = {
	fontWeight: 550,
	fontSize: 'clamp(1.25rem, 6vmin, 15rem)',
	color: 'rgb(0, 255, 255)',
	lineHeight: '1',
	textAlign: 'center',
	marginTop: 0,
	marginBottom: '0.6rem',
	textShadow: `
		0 0 5px rgba(0, 255, 255, 0.5),
		0 0 10px rgba(0, 143, 143, 0.5)
	`,
};

export const cardsWrapperStyle: CSSObject = {
	position: 'relative',
	width: '100%',
	height: '70vh',
	minHeight: '300px',
	overflow: 'hidden',
	marginBottom: '2rem',
};

export const cardStyle: CSSObject = {
	position: 'absolute',
	top: 0,
	left: 0,
	boxSizing: 'border-box',
	width: '100%',
	height: '100%',
	display: 'flex',
	flexDirection: 'column',
	justifyContent: 'space-between',
	transition: 'all 0.5s ease-in-out',
	background: 'linear-gradient(to bottom, rgba(0,255,255,0.1) 0%, rgba(0,255,255,0.05) 100%)',
	border: '1px solid rgba(255,255,255,0.08)',
	borderRadius: '6px',
	backdropFilter: 'blur(10px)',
	boxShadow: `
		inset 0 1px rgba(255,255,255,0.25),
		inset 0 -1px rgba(0,255,255,0.12)
	`,
	padding: '2rem',
	gap: '2rem',
};

export const mainContentStyle: CSSObject = {
	display: 'flex',
	flexDirection: 'row',
	alignItems: 'center',
	flexGrow: 1,
	gap: '2rem',
	[`@media (max-width: 600px)`]: {
		flexDirection: 'column',
	},
};

export const cardImageStyle: CSSObject = {
	height: '100%',
	maxHeight: '50vmin',
	objectFit: 'cover',
};

export const cardTextContainerStyle: CSSObject = {
	display: 'flex',
	flexDirection: 'column',
	justifyContent: 'center',
	height: '80%',
};

export const cardTechBubblesStyle: CSSObject = {
	width: '100%',
	display: 'flex',
	flexWrap: 'wrap',
	gap: '0.5rem',
	justifyContent: 'flex-start',
};

export const cardTitleStyle: CSSObject = {
	fontSize: '1.5rem',
	fontWeight: 600,
	color: 'transparent',
	background: 'linear-gradient(180deg, #dfe4ff 0%, #8c8ccf 100%)',
	backgroundClip: 'text',
	margin: '0 0 1rem 0',
	textAlign: 'left',
};

export const cardDescriptionStyle: CSSObject = {
	fontSize: '1rem',
	color: '#B6ADFF',
	lineHeight: '1.6',
	margin: '0 0 1rem 0',
	textAlign: 'left',
	textShadow: '0 0 2px rgba(182, 173, 255, 0.3)',
};

export const navigationContainerStyle: CSSObject = {
	display: 'flex',
	alignItems: 'center',
	justifyContent: 'center',
	gap: '2rem',
	width: '100%',
};

export const arrowButtonStyle: CSSObject = {
	background: 'rgba(182, 173, 255, 0.1)',
	border: '1px solid rgba(182, 173, 255, 0.1)',
	borderRadius: '50%',
	width: '48px',
	height: '48px',
	display: 'flex',
	alignItems: 'center',
	justifyContent: 'center',
	cursor: 'pointer',
	transition: 'all 0.3s ease',
	color: '#B6ADFF',

	'&:hover': {
		background: 'rgba(182, 173, 255, 0.2)',
		borderColor: 'rgba(182, 173, 255, 0.5)',
		transform: 'scale(1.05)',
	},

	'&:active': {
		transform: 'scale(0.95)',
	},
};

export const arrowIconStyle: CSSObject = {
	width: '20px',
	height: '20px',
};

export const dotsContainerStyle: CSSObject = {
	display: 'flex',
	gap: '0.75rem',
	alignItems: 'center',
};

export const dotStyle: CSSObject = {
	width: '12px',
	height: '12px',
	borderRadius: '50%',
	background: 'rgba(182, 173, 255, 0.3)',
	border: 'none',
	cursor: 'pointer',
	transition: 'all 0.3s ease',

	'&:hover': {
		background: 'rgba(182, 173, 255, 0.5)',
		transform: 'scale(1.2)',
	},
};

export const activeDotStyle: CSSObject = {
	background: 'linear-gradient(180deg, #dfe4ff 0%, #8c8ccf 100%)',
	boxShadow: '0 0 8px rgba(182, 173, 255, 0.5)',
	transform: 'scale(1.3)',
};
