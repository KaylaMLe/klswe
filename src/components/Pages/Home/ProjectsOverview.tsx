/** @jsxImportSource @emotion/react */
import React, { useState } from 'react';
import {
	slideshowContainerStyle,
	overviewTitleStyle,
	cardsWrapperStyle,
	cardStyle,
	cardContentStyle,
	cardTitleStyle,
	cardDescriptionStyle,
	navigationContainerStyle,
	arrowButtonStyle,
	arrowIconStyle,
	dotsContainerStyle,
	dotStyle,
	activeDotStyle,
} from './ProjectsOverview.styles.js';

interface ProjectCard {
	id: number;
	title: string;
	description: string;
	technologies: string[];
}

const projectCards: ProjectCard[] = [
	{
		id: 1,
		title: 'E-Commerce Platform',
		description:
			'A full-stack e-commerce solution built with React, Node.js, and PostgreSQL. Features include user authentication, payment processing, and inventory management.',
		technologies: ['React', 'Node.js', 'PostgreSQL', 'Stripe API'],
	},
	{
		id: 2,
		title: 'Task Management App',
		description:
			'A collaborative task management application with real-time updates, drag-and-drop functionality, and team collaboration features.',
		technologies: ['Vue.js', 'Socket.io', 'MongoDB', 'Express'],
	},
	{
		id: 3,
		title: 'Portfolio Website',
		description:
			'A personal portfolio website built with React, TypeScript, and Emotion. Features include a responsive design, smooth animations, and a contact form.',
		technologies: ['React', 'TypeScript', 'Emotion', 'Vite'],
	},
];

export function ProjectsOverview(): React.JSX.Element {
	const [currentSlide, setCurrentSlide] = useState(0);

	const nextSlide = () => {
		setCurrentSlide((prev) => (prev + 1) % projectCards.length);
	};

	const prevSlide = () => {
		setCurrentSlide((prev) => (prev - 1 + projectCards.length) % projectCards.length);
	};

	const goToSlide = (index: number) => {
		setCurrentSlide(index);
	};

	return (
		<div css={slideshowContainerStyle} id="projects-overview">
			<h2 css={overviewTitleStyle}>My work</h2>
			<div css={cardsWrapperStyle}>
				{projectCards.map((card, index) => (
					<div
						key={card.id}
						css={[
							cardStyle,
							{
								transform: `translateX(${(index - currentSlide) * 100}%)`,
								opacity: index === currentSlide ? 1 : 0.7,
							},
						]}
					>
						<div css={cardContentStyle}>
							<h3 css={cardTitleStyle}>{card.title}</h3>
							<p css={cardDescriptionStyle}>{card.description}</p>
							<div css={{ display: 'flex', flexWrap: 'wrap', gap: '0.5rem', marginTop: '1rem' }}>
								{card.technologies.map((tech) => (
									<span
										key={tech}
										css={{
											background: 'rgba(182, 173, 255, 0.2)',
											border: '1px solid rgba(182, 173, 255, 0.3)',
											borderRadius: '12px',
											padding: '0.25rem 0.75rem',
											fontSize: '0.875rem',
											color: '#B6ADFF',
										}}
									>
										{tech}
									</span>
								))}
							</div>
						</div>
					</div>
				))}
			</div>

			<div css={navigationContainerStyle}>
				<button css={arrowButtonStyle} onClick={prevSlide} aria-label="Previous project">
					<svg css={arrowIconStyle} viewBox="0 0 24 24" fill="none" stroke="currentColor">
						<path strokeLinecap="round" strokeLinejoin="round" strokeWidth={2} d="M15 19l-7-7 7-7" />
					</svg>
				</button>

				<div css={dotsContainerStyle}>
					{projectCards.map((_, index) => (
						<button
							key={index}
							css={[dotStyle, index === currentSlide && activeDotStyle]}
							onClick={() => goToSlide(index)}
							aria-label={`Go to slide ${index + 1}`}
						/>
					))}
				</div>

				<button css={arrowButtonStyle} onClick={nextSlide} aria-label="Next project">
					<svg css={arrowIconStyle} viewBox="0 0 24 24" fill="none" stroke="currentColor">
						<path strokeLinecap="round" strokeLinejoin="round" strokeWidth={2} d="M9 5l7 7-7 7" />
					</svg>
				</button>
			</div>
		</div>
	);
}
