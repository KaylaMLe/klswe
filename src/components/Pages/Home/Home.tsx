/** @jsxImportSource @emotion/react */
import React, { useState } from 'react';
import {
	pageStyle,
	hexagonBoxStyle,
	hexagonSvgStyle,
	heroTextStyle,
	nameStyle,
	titleStyle,
	subtitleStyle,
	aboutMeBoxStyle,
	glassBlurStyle,
	glassPanelStyle,
	aboutMeTextContainerStyle,
	aboutMeTitleStyle,
	aboutMeTextStyle,
} from './Home.styles';
import hexagonSvgUrl from '../../../assets/images/hexagon.svg';
import { GradientSweeps, StarBox } from './HomeBackground';
import { NavBar } from '../../NavBar/NavBar';

export default function Home(): React.JSX.Element {
	const [isMenuOpen, setIsMenuOpen] = useState(false);

	return (
		<div css={pageStyle}>
			<GradientSweeps />
			<NavBar isOpen={isMenuOpen} onToggle={() => setIsMenuOpen(!isMenuOpen)} />
			<StarBox />
			<Hexagon />
			<AboutMe />
		</div>
	);
}

function Hexagon(): React.JSX.Element {
	return (
		<div css={hexagonBoxStyle}>
			<img src={hexagonSvgUrl} alt="Decorative hexagon" css={hexagonSvgStyle} />
			<div css={heroTextStyle}>
				<h1 css={nameStyle}>Kayla Le</h1>
				<h2 css={titleStyle}>
					Full-Stack
					<br />
					Software Engineer
				</h2>
				<h3 css={subtitleStyle}>I build things for the web.</h3>
			</div>
		</div>
	);
}

function AboutMe(): React.JSX.Element {
	return (
		<div css={aboutMeBoxStyle} id="about-me">
			<div css={glassBlurStyle} />
			<div css={glassPanelStyle} />
			<div css={aboutMeTextContainerStyle}>
				<h2 css={aboutMeTitleStyle}>Hello!</h2>
				<p css={aboutMeTextStyle}>
					I'm a software engineer who thrives in fast-moving environments where ideas turn into products quickly. I'm
					happiest when the stakes are high and ideas are still taking shape.
				</p>
				<p css={aboutMeTextStyle}>
					I have experience in rapid prototyping: the art of turning rough ideas into working demos that bring concepts
					to life quickly. From hackathon projects to startup demos, I've tested feasibility, uncovered user value, and
					sparked bigger conversations.
				</p>
				<p css={aboutMeTextStyle}>
					I thrive in small teams where I can wear multiple hats and take ownership end-to-end. Startups energize me
					because they reward this exact creativity, adaptability, and initiative.
				</p>
			</div>
		</div>
	);
}
