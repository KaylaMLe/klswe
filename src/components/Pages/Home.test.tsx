import { fireEvent, render, screen } from '@testing-library/react';
import Home from './Home';
import { BrowserRouter } from 'react-router-dom';

describe('Home', () => {
	beforeEach(() => {
		render(
			<BrowserRouter>
				<Home />
			</BrowserRouter>
		);
		// mocks the click sound effect
		window.HTMLMediaElement.prototype.play = jest.fn();
	});

	it('renders the home page to fit the view port', () => {
		const pageContainer = screen.getByTestId('page-container');
		const navBar = screen.getByTestId('navbar');
		const mainContent = screen.getByTestId('main-content');

		expect(mainContent.offsetWidth).toEqual(pageContainer.offsetWidth);
		expect(mainContent.offsetHeight + navBar.offsetHeight).toEqual(pageContainer.offsetHeight);
	});

	it('adds a flower to the tree after interactions', () => {
		const tree = screen.getByAltText('Digital drawing of a bonsai tree');
		expect(tree).toBeInTheDocument();

		fireEvent.click(tree);
		// The alt text changes when a flower is added to the tree
		const oneFlower = screen.getByAltText(
			'Digital drawing of a bonsai tree with a light pink flower on it');
		expect(oneFlower).toBeInTheDocument();

		fireEvent.keyDown(tree, { key: 'Enter', code: 'Enter' });
		const twoFlowers = screen.getByAltText(
			'Digital drawing of a bonsai tree with 2 light pink flowers on it');
		expect(twoFlowers).toBeInTheDocument();

		fireEvent.keyDown(tree, { key: ' ', code: 'Space' });
		const threeFlowers = screen.getByAltText(
			'Digital drawing of a bonsai tree with 3 light pink flowers on it');
		expect(threeFlowers).toBeInTheDocument();
	});

	it('mutes the sound when the mute button is clicked', () => {
		const playSpy = jest.spyOn(window.HTMLMediaElement.prototype, 'play');

		const muteButton = screen.getByRole('button', { name: /Mute sound effects/ });
		fireEvent.click(muteButton);

		const tree = screen.getByAltText('Digital drawing of a bonsai tree');
		fireEvent.click(tree);

		expect(playSpy).not.toHaveBeenCalled();
	});
});