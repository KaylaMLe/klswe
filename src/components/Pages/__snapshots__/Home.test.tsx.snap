// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home renders properly on different display sizes 1`] = `
<div
  style="width: 1024px; height: 768px;"
>
  <div
    class="css-ccy2vc"
  >
    <nav
      class="css-zw4cqh"
    >
      <a
        class="css-42nl7f"
        href="#main"
      >
        Skip to main content
      </a>
      <div
        class="css-1iarqv4"
      >
        <a
          class="css-1x0xomp"
          href="/"
        >
          <h1
            class="css-dj9cgy"
          >
            Kayla Le
          </h1>
        </a>
        <h2
          class="css-1b38hje"
        >
          Software Engineer
        </h2>
      </div>
      <div
        class="css-1p063y5"
      >
        <a
          class="css-23kpuw"
          href="/about-me"
        >
          About me
        </a>
        <nav
          class="css-z6jdx8"
        >
          <button
            aria-controls="dropdown1"
            aria-expanded="false"
            aria-haspopup="listbox"
            class="css-4r3yzx"
          >
            Projects
            <div
              class="css-144hx5m"
              role="none"
            >
              ▼
            </div>
          </button>
        </nav>
      </div>
    </nav>
    <div
      class="css-v2kfba"
      id="main"
      role="none"
      tabindex="-1"
    >
      <div
        class="css-1cqxb9x"
      >
        <button
          class="css-1hiwl25"
        >
          <img
            alt=""
            src="test-file-stub"
          />
          Mute sound effects
        </button>
        <button
          aria-label="Press space or enter to play a chime and add a flower to the tree."
          class="css-17uo12l"
        >
          <audio
            id="clickSound"
          >
            <source
              src="test-file-stub"
            />
            Your browser does not support the audio element.
          </audio>
          <img
            alt="Digital drawing of a bonsai tree"
            class="css-v2kfba"
            src="test-file-stub"
          />
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`Home renders properly on different display sizes 2`] = `
<div
  style="width: 375px; height: 812px;"
>
  <div
    class="css-ccy2vc"
  >
    <nav
      class="css-zw4cqh"
    >
      <a
        class="css-42nl7f"
        href="#main"
      >
        Skip to main content
      </a>
      <div
        class="css-1iarqv4"
      >
        <a
          class="css-1x0xomp"
          href="/"
        >
          <h1
            class="css-dj9cgy"
          >
            Kayla Le
          </h1>
        </a>
        <h2
          class="css-1b38hje"
        >
          Software Engineer
        </h2>
      </div>
      <div
        class="css-1p063y5"
      >
        <a
          class="css-23kpuw"
          href="/about-me"
        >
          About me
        </a>
        <nav
          class="css-z6jdx8"
        >
          <button
            aria-controls="dropdown1"
            aria-expanded="false"
            aria-haspopup="listbox"
            class="css-4r3yzx"
          >
            Projects
            <div
              class="css-144hx5m"
              role="none"
            >
              ▼
            </div>
          </button>
        </nav>
      </div>
    </nav>
    <div
      class="css-v2kfba"
      id="main"
      role="none"
      tabindex="-1"
    >
      <div
        class="css-1cqxb9x"
      >
        <button
          class="css-1hiwl25"
        >
          <img
            alt=""
            src="test-file-stub"
          />
          Mute sound effects
        </button>
        <button
          aria-label="Press space or enter to play a chime and add a flower to the tree."
          class="css-17uo12l"
        >
          <audio
            id="clickSound"
          >
            <source
              src="test-file-stub"
            />
            Your browser does not support the audio element.
          </audio>
          <img
            alt="Digital drawing of a bonsai tree"
            class="css-v2kfba"
            src="test-file-stub"
          />
        </button>
      </div>
    </div>
  </div>
</div>
`;
